app/src/main/java/com/quran/labs/androidquran/ui/PagerActivity.java

```diff
@@ -526,6 +529,27 @@ public void onPageSelected(int position) {
     defaultNavigationBarColor = getWindow().getNavigationBarColor();
 
     quranEventLogger.logAnalytics(isDualPages, showingTranslation, isSplitScreen);
+
+    // Setup recitation (if enabled)
+    pagerActivityRecitationPresenter.bind(this, new PagerActivityRecitationPresenter.Bridge(
+        () -> numberOfPages,
+        () -> numberOfPagesDual,
+        () -> isDualPageVisible(),
+        () -> viewPager,
+        () -> audioStatusBar,
+        () -> ayahToolBar,
+        () -> slidingPagerAdapter,
+        () -> getSelectionStart(),
```

Comment:
why do we expose all of these? can we not just share the existing models and avoid directly sharing ui items?

Reply:
Refactored it to reduce the exposed ui items. Removed `numberOfPages`, `numberOfPagesDual`, `viewPager` and `slidingPagerAdapter` Reduced to:
```kotlin
  class Bridge(
    val isDualPageVisible: () -> Boolean,
    val currentPage: () -> Int,
    val audioStatusBar: () -> AudioStatusBar?,
    val ayahToolBar: () -> AyahToolBar?,
    val ensurePage: (ayah: SuraAyah) -> Unit,
    val showSlider: (sliderPage: Int) -> Unit,
  )
```
* AudioStatusBar is needed to set audiobar.listener, recitation enabled mode, and switch audiobar mode
* AyahToolBar is needed to set recitation enabled mode